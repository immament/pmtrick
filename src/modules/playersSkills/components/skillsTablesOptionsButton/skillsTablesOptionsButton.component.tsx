import React, { ReactNode } from 'react';
import { container } from 'tsyringe';

import { PlayersSkillsViewService } from '@src/modules/playersSkills/services/playersSkillsView.service';

import SkillsTableOptions from '../skillsTableOptionsModal/skillsTableOptions.component';

export class SkillsTablesOptionsButton extends React.Component<unknown, { show: boolean }> {
    state = {
        show: false,
    };

    private showModal = (_e: React.MouseEvent<HTMLButtonElement>): void => {
        this.setState({
            show: true,
        });
    };

    private handleClose = (_e: React.MouseEvent<HTMLButtonElement>): void => {
        this.setState({
            show: false,
        });
    };

    private handleSave = (): void => {
        const playersSkillsViewService = container.resolve(PlayersSkillsViewService);
        playersSkillsViewService.run();
    };

    render(): ReactNode {
        return (
            <React.Fragment>
                <button className="btn btn-default btn-success btn-xs" onClick={this.showModal}>
                    Options
                </button>
                <SkillsTableOptions show={this.state.show} onClose={this.handleClose} onSave={this.handleSave} />
            </React.Fragment>
        );
    }
}
