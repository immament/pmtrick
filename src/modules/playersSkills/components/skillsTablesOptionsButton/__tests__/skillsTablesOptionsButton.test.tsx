import 'reflect-metadata';

import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import { SkillsTablesOptionsButton } from '../skillsTablesOptionsButton.component';

describe('SkillsTableOptionsButton', () => {
    it('component renders', () => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
            .create(<SkillsTablesOptionsButton />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
