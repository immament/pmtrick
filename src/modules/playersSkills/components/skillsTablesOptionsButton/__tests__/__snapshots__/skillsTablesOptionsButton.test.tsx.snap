// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsTableOptionsButton component renders 1`] = `
<button
  className="btn btn-default btn-success btn-xs"
  onClick={[Function]}
>
  Options
</button>
`;
